doc/api/dns.md

```diff
@@ -417,6 +418,51 @@ is a two-dimensional array of the text records available for `hostname` (e.g.,
 one record. Depending on the use case, these could be either joined together or
 treated separately.
 
+## dns.resolveAny(hostname, callback)
+
+- `hostname` {string}
+- `callback` {Function}
+  - `err` {Error}
+  - `ret` {Object[][]}
+
+Uses the DNS protocol to resolve any queries (`ANY` records) for the `hostname`.
+The `ret` argument passed to the `callback` function will be an array of objects
+with uncertain type of records. Each object has a property `type` that indicates
+the type of current record. And for each type of record, the object structure
+will be like:
+
+| Type | Properties |
+|------|------------|
+| `"A"` | `address` / `ttl` |
+| `"AAAA"` | `address` / `ttl` |
+| `"CNAME"` | `value` |
+| `"MX"` | Refer to [`dns.resolveMx()`][] |
+| `"NAPTR"` | Refer to [`dns.resolveNaptr()`][] |
+| `"NS"` | `value` |
+| `"PTR"` | `value` |
+| `"SOA"` | Refer to [`dns.resolveSoa()`][] |
+| `"SRV"` | Refer to [`dns.resolveSrv()`][] |
+| `"TXT"` | This is an array-liked object with `length` and `indexes`, eg. `{'0':'sth','length':1}` |
```

Comment:
Because we need the extra column `type`.

You mean create an array with extra `type` directly?

eg.

```js
const arr = [ 'foo', 'bar' ];
arr.type = 'TXT';
```